src/main/java/jline/TerminalFactory.java

```diff
@@ -66,7 +66,7 @@ public static synchronized Terminal create(String ttyDevice) {
                 // it is ok to send ANSI control sequences
                 String emacs = System.getenv("EMACS");
                 String insideEmacs = System.getenv("INSIDE_EMACS");
-                if (emacs == null || insideEmacs == null) {
+                if (emacs == null && insideEmacs == null) {
                     type = NONE;
```

Comment:
This is not good. For emacs we need a mode that MUST NOT echo. Having ANSI control sequences are only secondary to that.

Reply:
Of the terminal types available, only the UnsupportedTerminal (which does not have ANSI) does not echo. Hence this PR should not be merged.